From 3a8f008715fb505a43bbd9b438ee6fc59264da52 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 15 Jan 2024 17:40:51 +0100
Subject: [PATCH 0578/1695] drivers: stm32_tamp: remove link to PSCI

This patch removes the direct call to psci_system_reset() in
tamper driver, because this function exist only on ARMv7 platform
STM32MP13 and STM32MP15. It don't exist for STM32MP25 platform.

This patch allows to support a system reset after TAMPER detection for
STM32MP25 platform by calling the new helper function do_reset().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 87b7aae859aa29f8b983b02d43281910a5d1c49e drivers: stm32_tamp: remove link to PSCI]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ic28aad9c8afa8f9d3635703b2f7ee687beaa825d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351404
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_tamp.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index 4550c7710..0c98119df 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -20,7 +20,6 @@
 #include <kernel/interrupt.h>
 #include <libfdt.h>
 #include <mm/core_memprot.h>
-#include <sm/psci.h>
 #include <stdbool.h>
 #ifdef CFG_STM32MP25
 #include <stm32_sysconf.h>
@@ -1389,10 +1388,8 @@ static enum itr_return stm32_tamp_it_handler(struct itr_handler *h __unused)
 				io_setbits32(base + _TAMP_SCR,
 					     _TAMP_SCR_ITAMP(id));
 
-#ifndef CFG_STM32MP25
 			if (ret & TAMP_CB_RESET)
-				psci_system_reset();
-#endif
+				do_reset("TAMPER");
 		}
 		i++;
 	}
@@ -1415,10 +1412,8 @@ static enum itr_return stm32_tamp_it_handler(struct itr_handler *h __unused)
 				io_setbits32(base + _TAMP_SCR,
 					     _TAMP_SCR_ETAMP(id));
 
-#ifndef CFG_STM32MP25
 			if (ret & TAMP_CB_RESET)
-				psci_system_reset();
-#endif
+				do_reset("TAMPER");
 		}
 		i++;
 	}
-- 
2.47.2

